src/components/v5/common/ActionSidebar/partials/ExpenditureBadge/ExpenditureBadge.tsx

```diff
@@ -1,43 +1,90 @@
 import clsx from 'clsx';
 import React, { type FC } from 'react';
+import { defineMessages } from 'react-intl';
 
-import { ExpenditureStatus } from '~gql';
 import Tooltip from '~shared/Extensions/Tooltip/index.ts';
 import { formatText } from '~utils/intl.ts';
 import PillsBase from '~v5/common/Pills/PillsBase.tsx';
 
 import { EXPENDITURE_STATE_TO_CLASSNAME_MAP } from './consts.ts';
-import { type ExpenditureBadgeProps } from './types.ts';
+import { useExpenditureActionStatus } from './hooks.ts';
+import {
+  ExpenditureActionStatus,
+  type ExpenditureBadgeProps,
+} from './types.ts';
 
-const ExpenditureBadge: FC<ExpenditureBadgeProps> = ({ status, className }) => {
-  const badgeTexts = {
-    [ExpenditureStatus.Draft]: formatText({ id: 'expenditure.draft' }),
-    [ExpenditureStatus.Cancelled]: formatText({ id: 'expenditure.cancelled' }),
-    [ExpenditureStatus.Finalized]: formatText({ id: 'expenditure.finalized' }),
-    [ExpenditureStatus.Locked]: formatText({ id: 'expenditure.locked' }),
-  };
+const displayName = 'v5.common.ActionSidebar.partials.ExpenditureBadge';
+
+const MSG = defineMessages({
+  badgeText: {
+    id: `${displayName}.badgeText`,
+    defaultMessage: `{activeStep, select,
+      ${ExpenditureActionStatus.Funding} {Funding}
+      ${ExpenditureActionStatus.Cancel} {Cancel}
+      ${ExpenditureActionStatus.Changes} {Changes}
+      ${ExpenditureActionStatus.Review} {Review}
+      ${ExpenditureActionStatus.Passed} {Paid}
+      ${ExpenditureActionStatus.Release} {Release}
+      ${ExpenditureActionStatus.Canceled} {Canceled}
+      ${ExpenditureActionStatus.Payable} {Payable}
+      other {Unknown}
+    }`,
+  },
+  tooltipText: {
+    id: `${displayName}.tooltipText`,
+    defaultMessage: `{activeStep, select,
+      ${ExpenditureActionStatus.Funding} {There is an active funding request for this payment.}
+      ${ExpenditureActionStatus.Review} {
+        Payment is currently in review. The payment creator can make changes freely until details are confirmed.
+      }
+      ${ExpenditureActionStatus.Changes} {
+        There is an active change request for this payment. Changes can be reviewed by clicking on the request.
+      }
+      ${ExpenditureActionStatus.Edit} {
+        The payment is currently in edit mode. After making the required changes, click the “Change payment”
+        button to confirm.
+      }
+      other {Unknown}
+    }`,
+  },
+});
+
+const ExpenditureBadge: FC<ExpenditureBadgeProps> = ({
+  expenditure,
+  className,
+  withAdditionalStatuses,
+}) => {
+  const activeStep = useExpenditureActionStatus(
```

Comment:
Perhaps this should be `activeStatus`?

Reply:
Agree with Jakub, `activeStatus` is a better name here